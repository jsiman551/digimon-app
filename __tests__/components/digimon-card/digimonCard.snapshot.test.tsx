import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import DigimonCard from '@/components/digimon-card';

jest.mock('next/image', () => ({
    __esModule: true,
    default: ({ src, alt, width, height }: { src: string; alt: string; width: number; height: number }) => (
        <img src={src} alt={alt} width={width} height={height} />
    ),
}))

describe('DigimonCard Snapshots', () => {
    const props = {
        name: 'Agumon',
        level: 'Champion',
        img: '/path/to/agumon.png',
    }

    it('matches the snapshot', () => {
        const { asFragment } = render(<DigimonCard {...props} />)
        expect(asFragment()).toMatchSnapshot()
    })
})
