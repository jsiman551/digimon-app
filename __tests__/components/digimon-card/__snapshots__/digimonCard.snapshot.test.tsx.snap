// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DigimonCard Snapshots matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="card bg-cyan-950 shadow-xl"
  >
    <figure>
      <img
        alt="Agumon"
        height="400"
        src="/path/to/agumon.png"
        width="400"
      />
    </figure>
    <div
      class="card-body"
    >
      <h2
        class="card-title mx-auto text-accent text-xl"
      >
        Agumon
      </h2>
      <p
        class="text-center text-white"
      >
        Level: 
        <span
          class="text-accent font-medium"
        >
          Champion
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
